src/apps/miscapps/h3ToGeoBoundaryHier.c

```diff
@@ -94,55 +97,90 @@ void recursiveH3IndexToGeo(H3Index h, int res, int isKmlOut) {
     }
 }
 
-int main(int argc, char* argv[]) {
-    // check command line args
-    if (argc < 2 || argc > 5) {
-        fprintf(stderr, "usage: %s H3Index [resolution outputMode]\n", argv[0]);
-        exit(1);
+int main(int argc, char *argv[]) {
+    int res;
+    H3Index parentIndex = 0;
+    char userKmlName[BUFF_SIZE] = {0};
+    char userKmlDesc[BUFF_SIZE] = {0};
+
+    Arg helpArg = {.names = {"-h", "--help"},
+                   .helpText = "Show this help message."};
+    Arg resArg = {.names = {"-r", "--resolution"},
+                  .scanFormat = "%d",
+                  .valueName = "res",
+                  .value = &res,
+                  .helpText =
+                      "Resolution, if less than the resolution of the parent "
+                      "only the parent is printed. Default 0."};
+    Arg parentArg = {
+        .names = {"-p", "--parent"},
+        .scanFormat = "%" PRIx64,
+        .valueName = "parent",
+        .value = &parentIndex,
+        .required = true,
+        .helpText = "Print cell boundaries descendent from this index."};
+    Arg kmlArg = {.names = {"-k", "--kml"},
+                  .helpText = "Print output in KML format."};
+    Arg kmlNameArg = {.names = {"--kn", "--kml-name"},
+                      .scanFormat = "%255c",  // BUFF_SIZE - 1
+                      .valueName = "name",
+                      .value = &userKmlName,
+                      .helpText = "Text for the KML name tag."};
+    Arg kmlDescArg = {.names = {"--kd", "--kml-description"},
+                      .scanFormat = "%255c",  // BUFF_SIZE - 1
```

Comment:
Worth assigning this to a constant or macro? Always needing to include the same explanatory comment is a bit of a code smell.